### PR TITLE
Adds the feature of Automatic Deployment to Github

### LINKED ISSUE #946: UX: Reset Loader when Zip is cancelled
Whenever the user click on cancel Uploading zip , the loader should reset.
## Steps to Reproduce
- Upload Zip and click generate app
- Cancel Zip when app is generated and upload new zip 

ui

### LINKED ISSUE #947: Automatic deployment of generated event website to the user's Github Account
I have been working on this feature for quite some time. This will give the users the ability to publish the event website on their Github account in the form of gh-pages.

**Workflow**
* Users will generate the event website with the help of the generator.
* If they want to deploy the generated website to their GitHub account, they will press on the **Deploy** button in the form.
* After clicking on the Deploy button, they will be taken to the **GitHub authentication** page where they will authorize the web app to make changes to their account.
* After they approve the requested permissions and enter their credentials, the process of uploading of files and making commits will start.
* After the whole website is completed, a **live link** to the gh-pages of the event website will appear on the page.

**Some Notes**
* I am still working on this feature. It would take some more time to get all the details right(interface, looks, error handling and display). But the core functionality has been somewhat implemented.
* Currently, the speed of uploading and committing of the file has been kept a bit low. On higher speeds, there were some conflicts since the Github database was not responding fast enough. I have emailed Github Support for an optimal no of requests per second.
* Authentication is managed through the library passport js.
* A repo named **eventSiteWebApp** is created in the user's account which contains the files of the event. User can delete that repository later if he wants.
* The app will ask for the user's permission to create and delete a repo from his/her account. 
* Since the process is a bit slow, please use the **EHealth Africa sample** for quick testing of the app.

Here is the link to the deployed app:
https://shrouded-ocean-89460.herokuapp.com
@mariobehling @aayusharora Please test and review. Thanks.

### PR DESCRIPTION
**Pull request by Princu7**
@aayusharora @mariobehling  Resolves #947 
Test Server: http://secure-meadow-20680.herokuapp.com

Please test the working of the server. Also, the feature won't work right away on the official server as we will have to do some modifications to the Heroku config variables and other things. Will do that after this gets reviewed :)

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -18,5 +18,9 @@
   },
   "proxy": "",
   "AWS_ACCESS_KEY_ID": "XXXXXXXXXXXXXXXXX",
-  "AWS_SECRET_ACCESS_KEY": "XXXXXXXXXXXXXXXXXXXXX"
+  "AWS_SECRET_ACCESS_KEY": "XXXXXXXXXXXXXXXXXXXXX",
```
Comment (aayusharora) [inline config.json]: @princu : We should add the info to update these keys in the readme.md. What you suggest ?

File: src/app.js [hunk 2]
```diff
@@ -68,11 +99,11 @@ io.on('connection', function(socket){
   socket.on('live', function(formData) {
     var req = {body: formData};
     generator.createDistDir(req, socket, function(appFolder, url) {
-      socket.emit('live.ready', {
-        appDir: appFolder,
-        url: url
-      });
-      //generator.showLivePreview(req, socket);
+        folder = appFolder;
```
Comment (aayusharora) [inline src/app.js]: @Princu7 
Why do we need to store a reference in variable called `folder` . Are we going to use it somewhere.

File: config.json (hunk 1 above)
Comment (geekyd) [inline config.json]: @aayusharora Yes we should have it on readme.md, will help new comers a lot.
@Princu7 What do you say?

Comment (Princu7) [inline config.json]: @aayusharora Yeah. There are many things(including these) which we should add on the README. I will add them in the next PR.

File: src/app.js (hunk 2 above)
Comment (Princu7) [inline src/app.js]: @aayusharora Actually, the folder variable was used to store the name of the folder whose contents were going to be uploaded and committed to Github. But it is not that good approach, so I did it other way.

**Comment by Princu7** [comment]
@aayusharora I have removed the merge conflicts. Also, please use the EHealth Sample zip file for testing the test server. Uploading and committing all the files to Github takes some time when we use the FossAsia 2016 sample zip since there are a lot of different files. Thanks!! :)

**Comment by mariobehling** [comment]
@aayusharora @Princu7 Can we merge this PR and improve things then step by step?

**Pull request merged**